internal/app/orchestrator/orchestrator.go

```diff
@@ -137,22 +132,17 @@ func (o *orchestrator) CreateWatch(req transport.Request) (transport.Watch, func
 
 	aggregatedKey, err := o.mapper.GetKey(req)
 	if err != nil {
-		// Can't map the request to an aggregated key. Log and continue to
-		// propagate the response upstream without aggregation.
+		// Can't map the request to an aggregated key. Log error and return.
 		o.logger.With(
 			"error", err,
 			"request_type", req.GetTypeURL(),
-			"request", req,
-		).Warn(ctx, "failed to map to aggregated key")
-		// Mimic the aggregated key.
-		// TODO (https://github.com/envoyproxy/xds-relay/issues/56). Can we
-		// condense this key but still make it granular enough to uniquely
-		// identify a request?
-		if req.GetRaw().V2 != nil {
-			aggregatedKey = fmt.Sprintf("%s%s", unaggregatedPrefix, req.GetRaw().V2)
```

Comment:
Can you check if the `GetRaw` function is still used in other places and remove it if its not.

Reply:
Still used here unfortunately: https://github.com/envoyproxy/xds-relay/blob/85bc48c1fffae9a13e7c41a5d572d6eeec48496a/internal/app/orchestrator/orchestrator.go#L163 :(

Reply:
i think this is a mistake from my previous refactoring. There should be handling for both v2/v3 here. I will fix it in a separate PR